components/cu_transform/src/velocity.rs

```diff
@@ -0,0 +1,357 @@
+use bincode::{Decode, Encode};
+use core::fmt::Debug;
+use serde::{Deserialize, Serialize};
+use uom::si::angular_velocity::radian_per_second;
+use uom::si::f32::AngularVelocity as AngularVelocity32;
+use uom::si::f32::Velocity as Velocity32;
+use uom::si::f64::AngularVelocity as AngularVelocity64;
+use uom::si::f64::Velocity as Velocity64;
+use uom::si::velocity::meter_per_second;
+
+/// A struct representing linear and angular velocity in 3D space
+///
+/// The `VelocityTransform` encapsulates both linear and angular velocity components
+/// of a rigid body in 3D space. It complements the `Transform3D` struct which
+/// represents position and orientation.
+///
+/// Linear velocity is represented as a 3D vector [vx, vy, vz] in meters per second.
+/// Angular velocity is represented as a 3D vector [wx, wy, wz] in radians per second,
+/// where the vector represents the axis of rotation and the magnitude is the rate.
+///
+/// # Examples
+///
+/// ```
+/// use cu_transform::VelocityTransform;
+///
+/// // Create a velocity with 1 m/s forward and 0.5 rad/s rotation around Z axis
+/// let velocity = VelocityTransform::<f32> {
+///     linear: [1.0, 0.0, 0.0],
+///     angular: [0.0, 0.0, 0.5],
+/// };
+/// ```
+///
+/// Velocities can be transformed between coordinate frames using the
+/// `transform_velocity` function, and new velocities can be computed from time-stamped
+/// transforms using `StampedTransform::compute_velocity`.
+///
+/// When using the TransformTree to look up velocities with `lookup_velocity`, results
+/// are automatically cached for improved performance on repeated lookups. The cache is
+/// invalidated when new transforms are added or when entries exceed their age limit.
+#[derive(Debug, Clone, Encode, Decode, Serialize, Deserialize)]
+pub struct VelocityTransform<T: Copy + Debug + 'static> {
+    /// Linear velocity components [vx, vy, vz] in meters per second
+    pub linear: [T; 3],
+    /// Angular velocity components [wx, wy, wz] in radians per second
+    pub angular: [T; 3],
+}
+
+impl VelocityTransform<f64> {
```

Comment:
maybe use https://docs.rs/num-traits/latest/num_traits and parametrize over all the compatible number formats?